python/plugins/processing/algs/qgis/PointsToPaths.py

```diff
@@ -165,7 +165,8 @@ def processAlgorithm(self, parameters, context, feedback):
 
         current = 0
         total = 100.0 / len(points) if points else 1
-        for group, vertices in list(points.items()):
+        verlist = list(points.items()) # (2)
```

Comment:
what would be the optimization here? I can't figure it out, sorry!

Reply:
I think the problem here is calling `list(dict.items())` -- that's an anti pattern. Do you see the same results if you change this just to:

    for group, vertices in points.items():